harness/determined/keras/callbacks.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict
+
+import tensorflow as tf
+
+
+class Callback(tf.keras.callbacks.Callback):  # type: ignore
+    """
+    A Determined implementation of Callback which supports additional new callbacks.
+    """
+
+    def on_validation_period_begin(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_validation_period_end(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_train_step_begin(self, total_batches_trained: int, logs: Dict) -> None:
```

Comment:
question: why are exposing `train_step` since that is more an internal concept now?

Reply:
This is the only way to make sure you get all of the training metrics.  See the `ConfusionMatrixWriter` callback example in the RFC doc.

I'm totally open to a better name though.